draft-wilton-netconf-yp-observability.md

```diff
@@ -84,15 +92,105 @@ For other systems, the operational information may be distributed across multipl
 
 Currently, it:
 
-- defines a new YANG-Push encoding format that can be used for both on-change and periodic subscriptions that reports the data from the subscription filter point.
+- Uses {{RFC8639}} to define what a subscription is and how to create it.  This includes the terminology defined in this RFC such as  configured subscriptions and dynamic subscriptions [TBD]
+- Replaces {{RFC8641}} [TBD]
+- Defines a single message type for both 'on change' and periodic updates
+- Provides an extensible option for data encoding
+- Provides the concept of a heartbeat interval to 'on change' subscripitons
+- Adds a new subscripiton type named periodic-on-change and describes the differences between this and on-change with heartbeat
+- Provides an end-of-sync marker
+- Clarifies the behaviour when authorization changes (local and remote)
+- Removes the YANG patch format
+- Mandates the use of the YANG push envelope header as defined in {{I-D.draft-netana-netconf-notif-envelope}}
+- Provides a subscription-path option in addition to a path option [TBD]
+- Removes the negotiation phase when setting up subscriptions replacing it with a simple accept/reject at that instantanous point in time
+- Describes that only "change" and "delete" updates are sent
+
+
+
+## Single message type for on-change and periodic updates
+
+The original YANG push drafts define two message types for updates ("push-update" and "push-change-update").
+
+Collectors do not need to identify update messages between stream updates and on-change updates, as such a unified
+message type called "push-update-notification" is defined that is provided for all updates no matter what the
+subscription type is.
+
+
+## Provides an extensible option for data encoding
+
+NETCONF is an XML based protocol and all NETCONF operations and notifications are defined in XML.  Inside NETCONF
+operations and notifications is encoded data.  This encoded data can be delivered in multiple encodings, there need
+not necessarily be a limitation on what encoding this data takes as long as the server/producer can encode it and the
+client/receiver can decode it.
+
+There are a number of common data encodings in use today (and already standardised by the IETF), however, there may
+be new encodings in the future.  This draft, thereforem, will provide an extensible encoding selection to allow
+for future additions.
+
+The three data encodings provided initially in this draft are:
+
+- XML as defined in {{RFC7950}}
+- JSON_IETF as defined in {{RFC7951}}
+- CBOR as defined in {{RFC9254}}
+
+This draft does not mandate any specific data encoding nor does it define the default encoding, instead, leaving the
+encoding selection of choice to the consumer (or producer in the case of a configured subscription).
+
+The establish-subscription operation is extended with an "encoding" option to determine the encoding type.  This "encoding" leaf is an enumeration with the following options: xml, json_ietf, cbor
```

Comment:
```suggestion
The establish-subscription operation is extended with an "encoding" option to determine the encoding type.  This "encoding" leaf is an identityref with the following defined identities: xml, json_ietf, cbor
```
This enables better extensibility (should one choose to augment additional encodings at a later point)